plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/BaseOpenSearchConnectorTest.java

```diff
@@ -1883,6 +1924,410 @@ public void testQueryTableFunction()
+            index(tableName, record3);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        String selectQuery = "SELECT id, root.f1 FROM " + tableName;
+        String expectedResult = "VALUES (BIGINT '1', BIGINT '1'), (BIGINT '2', NULL), (BIGINT '3', NULL)";
+
+        // With Projection Pushdown enabled
+        assertThat(query(selectQuery))
+                .matches(expectedResult)
+                .isFullyPushedDown();
+
+        // With Projection Pushdown disabled
+        Session sessionWithoutPushdown = sessionWithProjectionPushdownDisabled(getSession());
+        assertThat(query(sessionWithoutPushdown, selectQuery))
+                .matches(expectedResult)
+                .isNotFullyPushedDown(ProjectNode.class);
+
+        try {
+            deleteIndex(tableName);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    @Override // Override as connector doesn't support creating tables.
+    public void testProjectionWithCaseSensitiveField()
+    {
+        String tableName = "test_projection_with_case_sensitive_field_" + randomNameSuffix();;
+        @Language("JSON")
+        String properties =
+                """
+                {
+                    "properties": {
+                        "id": {
+                            "type": "integer"
+                        },
+                        "a": {
+                            "properties": {
+                                "UPPER_CASE": {
+                                    "type": "integer"
+                                },
+                                "lower_case": {
+                                    "type": "integer"
+                                },
+                                "MiXeD_cAsE": {
+                                    "type": "integer"
+                                }
+                            }
+                        }
+                    }
+                }
+                """;
+
+        try {
+            createIndex(tableName, properties);
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 1L)
+                    .put("a", ImmutableMap.<String, Object>builder()
+                            .put("UPPER_CASE", 2)
+                            .put("lower_case", 3)
+                            .put("MiXeD_cAsE", 4)
+                            .buildOrThrow())
+                    .buildOrThrow());
+            index(tableName, ImmutableMap.<String, Object>builder()
+                    .put("id", 5L)
+                    .put("a", ImmutableMap.<String, Object>builder()
+                            .put("UPPER_CASE", 6)
+                            .put("lower_case", 7)
+                            .put("MiXeD_cAsE", 8)
+                            .buildOrThrow())
+                    .buildOrThrow());
+        }
+        catch (IOException e) {
```

Comment:
Don't catch and rethrow. Just add the exception to the method declaration. Same comment for all other tests.

Reply:
These methods are overridden from BaseConnectorTest

Reply:
Created separate methods in class instead of overriding from `BaseConnectorTest`.